Code-4-Community/speak-for-the-trees-frontend
# NavBar Improvements

## Why

[ClickUp Ticket](https://app.clickup.com/t/2nf5hx4)



End user:
Adds a link to the My Trees page to the nav bar dropdown menu to make navigation easier. Fixes some related logout errors.

Developers:
Improves readability and organization of navbar components.

## This PR



- Makes NavBar a "dumb" component, following our convention such that only containers and App are "smart."
- To improve readability, creates subcomponents for NavBar which are also shared with MobileNavBar. 
  - NavBar for overall navbar that renders content according to window size
  - NavExtra for content depending on whether or not the user is logged in
  - NavMenu for the dropdown menu, common to NavExtra and MobileNavBar
- Adds the My Trees link to NavMenu
- Create a global history to resolve some logout errors (was experiencing an issue where logout didn't always work), use this history to force reload.
- Some styling changes to look prettier.
- Reorganizes all navbar-related files into the navBar directory.

## Screenshots

 

NavBar (desktop, admin)
![image](https://user-images.githubusercontent.com/22990100/160960553-300a3e13-e92c-409a-8297-8ddbd32f43a9.png)
![image](https://user-images.githubusercontent.com/22990100/160960575-49427072-cdf2-404b-ab86-39f52a2291fc.png)

NavBar (mobile, not admin)
![image](https://user-images.githubusercontent.com/22990100/160960635-60f8b8b2-6b9d-44f5-8596-fc9febf9a1b8.png)

## Verification Steps

 
Created some tests to ensure the proper content renders. Checked to make sure things looked reasonable and logout worked on Chrome, Safari, and Firefox (Mac).
